app/src/main/java/com/termux/app/utils/PluginUtils.java

```diff
@@ -0,0 +1,332 @@
+        // to show the details of the error
+        String title = TermuxConstants.TERMUX_APP_NAME + " Plugin Execution Command Error";
+
+        StringBuilder reportString = new StringBuilder();
+
+        reportString.append(ExecutionCommand.getExecutionCommandMarkdownString(executionCommand));
+        reportString.append("\n\n").append(TermuxUtils.getAppInfoMarkdownString(context, true));
+        reportString.append("\n\n").append(TermuxUtils.getDeviceInfoMarkdownString(context));
+
+        Intent notificationIntent = ReportActivity.newInstance(context, new ReportInfo(UserAction.PLUGIN_EXECUTION_COMMAND, logTag, title, null, reportString.toString(), null,true));
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        // Setup the notification channel if not already set up
+        setupPluginCommandErrorsNotificationChannel(context);
+
+        // Use markdown in notification
+        CharSequence notificationText = MarkdownUtils.getSpannedMarkdownText(context, executionCommand.errmsg);
+        //CharSequence notificationText = executionCommand.errmsg;
+
+        // Build the notification
+        Notification.Builder builder = getPluginCommandErrorsNotificationBuilder(context, title, notificationText, notificationText, pendingIntent, NotificationUtils.NOTIFICATION_MODE_VIBRATE);
+        if (builder == null)  return;
+
+        // Send the notification
+        int nextNotificationId = NotificationUtils.getNextNotificationId(context);
+        NotificationManager notificationManager = NotificationUtils.getNotificationManager(context);
+        if (notificationManager != null)
+            notificationManager.notify(nextNotificationId, builder.build());
+
+    }
+
+    /**
+     * Send {@link ExecutionCommand} result {@link PendingIntent} in the
+     * {@link TERMUX_SERVICE#EXTRA_PLUGIN_RESULT_BUNDLE} bundle.
+     *
+     *
+     * @param context The {@link Context} that will be used to send result intent to the {@link PendingIntent} creator.
+     * @param logTag The log tag to use for logging.
+     * @param label The label of {@link ExecutionCommand}.
+     * @param stdout The stdout of {@link ExecutionCommand}.
+     * @param stderr The stderr of {@link ExecutionCommand}.
+     * @param exitCode The exitCode of {@link ExecutionCommand}.
+     * @param errCode The errCode of {@link ExecutionCommand}.
+     * @param errmsg The errmsg of {@link ExecutionCommand}.
+     * @param pluginPendingIntent The pluginPendingIntent of {@link ExecutionCommand}.
+     * @return Returns {@code true} if pluginPendingIntent was successfully send, otherwise [@code false}.
+     */
+    public static boolean sendPluginExecutionCommandResultPendingIntent(Context context, String logTag, String label, String stdout, String stderr, Integer exitCode, Integer errCode, String errmsg, PendingIntent pluginPendingIntent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.